### PR TITLE
follow up pr to instant onboarding  & add as second device scanner improvements

### LINKED ISSUE #3676: windows: add-second-device issues
i recently got the opportunity to watch someone using "add second device" from android to desktop; there are the issues that pop up:

- for whatever reason, the QR code scanner rectangle is larger than it fits on the screen, although the window was full-height. this leads to the what-to.do-hints not being visible:

image_2024-02-10_19-40-36

with just that, the user is pretty lost when starting the process from the new device. the hint **"On the first device, go to "Settings / Add Second Device" and scan the code shown there"** is only shown when the user discovers that there can be sth. scrolled

on comparable large mac screens (the window screen was a normal 15~16" thinkpad laptop), the hint is shown, at least in full-height windows (it still disappears when the window is shrinked)

it seems better to never add scrollbars there but to scale the QR code scanner rectangle and make hint being always visible (the overlay in the camera window is nice, btw)

2. in contrast to mac, the camera image is not acting in the same way as a mirror. this leads to sort of intelligence test when moving the qr code around. this is a minor thing, it adds some initial confusion though, where ppl also have to deal with contrast and distance. if it gets complicated to fix that, it is maybe not worth fixing.
 
the detection of the QR code as such is pretty nice and did not cause issues

the used laptop was a standard lenovo one with a standard out-of-the-box windows, nothing tweaked.

finally, there was https://support.delta.chat/t/add-hint-select-on-windows-desktop-private-network-to-add-a-second-device/2921/2 , but i fixed that before and i am working on a PR to get that to the "troubleshooting" (maybe we should get that to the primary windows UI as well, however, with an iroh update, there may also be holepunching, so starting with "troubleshooting" seems fine)

### LINKED ISSUE #3799: DCLOGIN broken when logged in
It does not work when you scan a dclogin code in a configured account.

This bug was introduced by merging #3773

### LINKED ISSUE #3803: Help does not open for missing languages
actual: help does not open

expected: help should open in the English version

error:
```
node:16376) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, stat '~/Coding/deltachat-desktop/html-dist/help/nl_NL/help.html'
(node:16376) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
```

### PR DESCRIPTION
**Pull request by Simon-Laux**
- fixes #3799
- also add steps and troubleshooting link to " add as second device qr code scan" dialog #3178
- fix help not opening for languages that have no localised help

closes #3676
closes #3803

### CODE REVIEW

**Comment by r10s** [comment]
thanks a lot for taking care! ❤️

some high-level comments:

- it seems fine to regard the "My Profile" screen as a confirmation dialog, indeed, nothing has happend if the user taps "back". thanks for thinking that over!

- the new "troubleshooting" button is fine, indeed, as you said, "more options" is gone, so it's better to have that directly there

- for the new anchor [#multiclient-troubleshooting](https://delta.chat/en/help#troubleshooting) vs. [#multiclient](https://delta.chat/en/help#multiclient), i do not think it is _that_ clear, which is better, regarding the current help. if you do not know much about "add second device" at all, a little introduction is needed, and counts to troubleshooting as well, even if the headline is not literally correct,  the "Troubleshooting" immediately follows. maybe we should have named the button just "Help" and/or organize the paragraphs differently.  
for the sake of ease and consistency, however, for now, let's stay with the anchor and the button label. at some point we will iterate over the section anyways, no need to change things before (also as adding anchors adds noise to translations etc.)  
esp, as there was just nothing before :)

- for the "add second device" screen, i'd unclutter the above area as follows:

**Comment by Simon-Laux** [comment]
Sure the troubleshooting section is visible, but not that much, otherwise I wouldn't have raised it as an issue:
Bildschirmfoto 2024-05-02 um 00 13 20

Though we can say that people looking for it will surely find it (provided the other sections above it do not grow too much (content updates and. other languages might need less or more characters/space to express the concepts).

About renaming to "help", I wouldn't do that because of the following reasoning: To give some stereotype, some group of people try around stuff and would never "read the manual" first. I have the feeling that the word "troubleshooting" is less of a problem than admitting to needing "help" for those people. (I personally also have that tendency of trying out things before reading the manual, though I don't know if that is an effect of genetics/biology, my soul, the society/patriarchy or whatever).
But my position on this aspect is debatable, you could also argue that more people want assistant to understand what second device setup is at all, though I doubt that many people would stumble across that dialog by accident.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,7 +9,8 @@
 - add quick key `Cmd+W`/`Ctrl+W` to close webxdc-, html_email- and help-window #3770 #3778
 - Accept images from clipboard in QR reader #3762
 - Introduce new `Spinner` component #3786
-- Instant Onboarding #3773
+- Instant Onboarding #3773 #3801
+- Add steps to add as second device qr code scan dialog #3801
```
Comment (r10s) [inline CHANGELOG.md:13]: this is hard to understand, maybe the following:

```suggestion
- Add instructions and troubleshooting button to "Add as Second Device" dialog #3801
```

File: src/renderer/hooks/useInstantOnboarding.ts [hunk 2]
```diff
@@ -81,22 +74,33 @@ export default function useInstantOnboarding(): InstantOnboarding {
...
-      // Use custom chatmail instance if given by QR code
-      if (welcomeQr && welcomeQr.qr.kind === 'account') {
-        instanceUrl = welcomeQr.url
+      let configurationQR = `dcaccount:${DEFAULT_CHATMAIL_QR_URL}`
+
+      if (welcomeQr) {
+        // Use custom chatmail instance if given by QR code
+        if (welcomeQr.qr.kind === 'account') {
+          // 1. In this "Instant Onboarding" account creation flow the user is not
+          // asked to manually insert any mail server credentials.
```
Comment (r10s) [inline src/renderer/hooks/useInstantOnboarding.ts:83]:
```suggestion
          // asked to manually enter any mail server credentials.
```

**Review by r10s — APPROVED** [review]
yeah, this is very nice improvement wrt layout and functionality now 👌

code lgtm - as far as i can tell :)

**Pull request merged**